dupes.sh

```diff
@@ -45,29 +45,30 @@ fi
 # the fingerprints of all student submissions
 declare -A fingerprints
 
-# the fingerprints of the whitelisted files
-declare -A whitelist
+# the fingerprints of the nicelisted files
+declare -A nicelist
 
 # enable recursive globbing **/*.java
 shopt -s globstar
 
-# for now, exit if plagiarism whitelist directory does not exist
-test -e "$WHITELISTDIR" || { echo "plagiarism whitelist does not exist"; exit 1; }
+# for now, exit if plagiarism nicelist directory does not exist
+test -e "$NICELISTDIR" || { echo "plagiarism nicelist does not exist"; exit 1; }
 
-for file in "$WHITELISTDIR"/**/*.java; do
-    # if there are no whitelisted files we get the glob expression, because shell programming
-    test -e "$file" || break
-    whitelist[`fingerprint "$file"`]="$file"
+for file in "$NICELISTDIR"/**/*; do
+    # if there are no nicelisted files we get the glob expression, because shell programming
+    # also we glob directories, which we also want to avoid checking
+    test -f "$file" || break
+    nicelist[`fingerprint "$file"`]="$file"
 done
 
 echo Dupechecking
 for arg in "$@"; do
-    for file in "$arg"/**/*.java; do
-        test -e "$file" || break
-        test -e "${file}.SUSPECT" && continue
+    for file in "$arg"/**/*; do
+	#check we are indeed dealing with a file and it is not an archive contents list
+        test -f "$file" && [ "${file: -9}" != ".contents" ] || continue
```

Comment:
With the glob expression **/* you might get all kinds of unwanted files, which you then exclude here. What about binary files, or IDE project files? These should also be excluded. I understand that hard-coding **/*.java is not what we want. Can you make a configurable list of file extensions that should be globbed for?

Reply:
Alright, so I have implemented this. Learned some more things about bash programming on the way. I also added tests, since there were some constructs (eval, cbn array passing, iteration with while) that I needed a while to figure out, therefore I wanted to make sure that the changes were test-documented. It would be nice to have some sort of CI I guess also.

I had to refactor the functions out of `dupes.sh`, else I would not have been able to test my added feature in isolation.

Reply:
Very nice! One more thing: In initNiceList you only search for files with the given extensions. This should be done here. There still is the glob `**/*`. Can you do that?

Reply:
Alright, I've done that now, also ran it through a system test by using it for this week's submissions.

Reply:
Nice! I'm merging your patches.